syntax = "proto3";

// This defines the protocol buffers used by XBOS with WaveMQ. The schema for
// a wavemq xbos proto message is:
//   xbosproto/RootMessageType
//
// Version 1.0
package xbospb;


// Hamilton Project definitions
import "hamilton.proto";

// XBOS IoT device definitions
import "iot.proto";

// Dent meters
import "dentmeter.proto";

// System monitors
import "system_monitor.proto";

// Root union type
// We reserve 50 fields per imported file
message XBOS {

  //Hamilton Fields
  HamiltonData HamiltonData = 50;
  HamiltonBRLinkStats HamiltonBRLinkStats = 51;
  HamiltonBRMessage HamiltonBRMessage = 52;

  //IoT Fields
  XBOSIoTDeviceState XBOSIoTDeviceState = 100;
  XBOSIoTDeviceActuation XBOSIoTDeviceActuation = 101;
  XBOSIoTContext XBOSIoTContext = 102;

  //Dent Fields
  DentMeterState DentMeterState = 150;

  //System Status Fields
  BasicServerStatus BasicServerStatus = 200;
}
